client/my-sites/pages/page-list.jsx

```diff
@@ -202,7 +213,7 @@ var Pages = React.createClass( {
 			rows = [];
 
 		// pages have loaded, sites have loaded, and we have a site instance or are viewing all-sites
-		if ( pages.length && this.props.sites.initialized ) {
+		if ( pages.length && this.props.hasSites ) {
```

Comment:
@mcsf -- you suggested that I might be able to do something like this to avoid the need for the new state value:

```			<PostListFetcher
				type="page"
				siteID={ this.props.hasSites ? this.props.siteId : undefined }
```

...where `this.props.hasSites` is defined as `! isEmpty( state.sites.items )`.

Not a bad thought & it solves the flash of all sites pages from showing on a cleared out indexeddb.

Unfortunately, switching to "all sites" afterward never triggers a render & the loading indicators stick around (even though a `/v1.1/me/posts?http_envelope=1&status=trash&order_by=date&order=DESC&type=page&site_visibility=visible` XHR is completed during the initial page load.

Let me know if you can think of anything else I should try to avoid the new piece of state & selector here or if you think this can make it in as-is!  Thank you!